fedimint-core/src/core.rs

```diff
@@ -70,17 +73,33 @@ impl OperationId {
     pub fn from_encodable<E: Encodable>(encodable: &E) -> OperationId {
         Self(encodable.consensus_hash::<sha256::Hash>().to_byte_array())
     }
+
+    pub fn fmt_short(&self) -> OperationIdShortFmt {
+        OperationIdShortFmt(self)
+    }
+    pub fn fmt_full(&self) -> OperationIdFullFmt {
+        OperationIdFullFmt(self)
+    }
+}
+
+impl<'a> Display for OperationIdShortFmt<'a> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        bitcoin29::hashes::hex::format_hex(&self.0 .0[0..4], f)?;
+        f.write_str("_")?;
+        bitcoin29::hashes::hex::format_hex(&self.0 .0[28..], f)?;
```

Comment:
do we even care about the suffix?

Reply:
Good question. I kind of wanted to make sure that the whole thing is longer and just got shortened, I guess, so "removing the middle" seemed to make sense. And if people are going to compare, they'll visually match beginning or the end. But no strong feelings about it. I guess `abcdefgh_` or even just `abcdefgh` would work.

Reply:
I think prefix is enough, we just want different ids for different operations

Reply:
Are `OperationId`s always hashes thought? Could there be a case where lots of them have the same prefix for some reason? @maan2003

Reply:
I think they are always hashes now that recovery is not a state machine. But yes external module developer can implement it however they like

Reply:
then it is fine with prefix and suffix